hugophotoswipe/photo.py

```diff
@@ -105,13 +108,105 @@ def _load_original_image(self):
         # fallback for unhandled rotation tags
         return img
 
+    def _load_exif(self, image):
+        if settings.exif:
+            tags = set(_filter_tags(TAGS.values(),
+                                    settings.exif.get('include'),
+                                    settings.exif.get('exclude')))
+            tags.add('Orientation')  # always need this for resize
+        else:
+            tags = TAGS.values()
+
+        exif_data = {}
+        exif = image._getexif() or {}
+        for k, v in exif.items():
+            decoded = TAGS.get(k)
+            if decoded in tags and not isinstance(v, IFDRational):  # Filter complex data values
+                exif_data[decoded] = v
+        for k, v in exif_data.pop('GPSInfo', {}).items():
+            decoded = GPSTAGS.get(k, k)
+            exif_data[decoded] = v
+        self._exif = exif_data
+
     def free(self):
         """Manually clean up the cached image"""
         if hasattr(self, "_original_img") and self._original_img:
             self._original_img.close()
             del self._original_img
         self._original_img = None
 
+    @property
+    def iptc(self):
```

Comment:
What's the rationale for not having a `_load_iptc` method, similar to the `_load_exif` method?

Reply:
Good point & implemented. 

The only downside is that it now always attempts to load IPTC. We always needed exif for orientation but we could lazy-load IPTC on first use. Right now it will simply load both.

The benefit of the code alignment may outweigh the performance penalty for users that don't use iptc data. I haven't tested the performance penalty yet.

Reply:
I'm not sure I understand why you can't lazy load IPTC anyway. In `_get_tag_value`, you can load IPTC when needed.

Instead of:
```
o = getattr(self, obj.lower(), {})
```
you could use:
```
if obj.lower() == 'exif':
  return self.exif.get(t)
return self.iptc.get(t)
```
Then, in `def iptc(self)`, you can have:
```
if self._iptc is None:
  self._load_iptc()
return self._iptc
```

This means you won't have `_load_iptc` in the same place as `_load_exif`, but I think that's acceptable.

Let me know if you're interested in making this change. This has been open for a long time so if you're not, we could also just merge it and I can handle it instead.